index.js

```diff
@@ -200,9 +275,250 @@ function fastifyMultipart (fastify, options, done) {
+      onError(new FieldsLimitError())
+    })
+
+    request.pipe(bb)
+
+    function onField (name, fieldValue, fieldnameTruncated, valueTruncated) {
+      // don't overwrite prototypes
+      if (getDescriptor(Object.prototype, name)) {
+        onError(new PrototypeViolationError())
+        return
+      }
+
+      const value = {
+        fieldname: name,
+        value: fieldValue,
+        fieldnameTruncated,
+        valueTruncated,
+        fields: body
+      }
+
+      if (body[name] === undefined) {
+        body[name] = value
+      } else if (Array.isArray(body[name])) {
+        body[name].push(value)
+      } else {
+        body[name] = [body[name], value]
+      }
+
+      ch(value)
+    }
+
+    function onFile (name, file, filename, encoding, mimetype) {
+      // don't overwrite prototypes
+      if (getDescriptor(Object.prototype, name)) {
+        // ensure that stream is consumed, any error is suppressed
+        sendToWormhole(file)
+        onError(new PrototypeViolationError())
+        return
+      }
+
+      const value = {
+        fieldname: name,
+        filename,
+        encoding,
+        mimetype,
+        file,
+        fields: body,
+        _buf: null,
+        buffer () {
+          if (this._buf) {
+            return Promise.resolve(this._buf)
+          }
+          return new Promise((resolve, reject) => {
+            pump(this.file, concat((buf) => {
+              this._buf = buf
+              resolve(this._buf)
+            })).catch(reject)
+          })
+        }
+      }
+      if (body[name] === undefined) {
+        body[name] = value
+      } else if (Array.isArray(body[name])) {
+        body[name].push(value)
+      } else {
+        body[name] = [body[name], value]
+      }
+
+      ch(value)
+    }
+
+    function onError (err) {
+      lastError = err
+    }
+
+    function onEnd (error) {
+      cleanup()
+      bb.removeListener('finish', onEnd)
+      bb.removeListener('error', onEnd)
+      ch(error || lastError)
+    }
+
+    function cleanup () {
+      // keep finish listener to wait all data flushed
+      // keep error listener to wait stream error
+      request.removeListener('close', cleanup)
+      bb.removeListener('field', onField)
+      bb.removeListener('file', onFile)
+      bb.removeListener('close', cleanup)
+    }
+
+    return parts
   }
+
+  async function handlePartFile (part, logger) {
+    const file = part.file
+    if (file.truncated) {
+      // ensure that stream is consumed, any error is suppressed
+      await sendToWormhole(file)
+      // throw on consumer side
+      return Promise.reject(new RequestFileTooLargeError())
+    }
+
+    file.once('limit', () => {
+      const err = new RequestFileTooLargeError()
+
+      if (file.listenerCount('error') > 0) {
+        file.emit('error', err)
+        logger.warn(err)
+      } else {
+        logger.error(err)
+        // ignore next error event
+        file.on('error', () => { })
```

Comment:
why is this ignored? Maybe we should log it as debug?

Reply:
I will log it as an error.